Source/WebCore/rendering/Grid.cpp

```diff
@@ -186,6 +186,12 @@ GridSpan Grid::gridItemSpan(const RenderBox& gridItem, GridTrackSizingDirection
     return direction == ForColumns ? area.columns : area.rows;
 }
 
+void Grid::setupGridForMasonryLayout()
+{
+    m_grid.clear();
```

Comment:
We should work on optimizing this, so we don't need to clear out the entire grid, because we ultimately want to keep the first row. Probably want to add a TODO here.

Reply:
Good idea. I'd file a bugzilla, and reference that bug # here.